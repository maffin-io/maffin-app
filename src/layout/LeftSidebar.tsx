import React from 'react';
import Link from 'next/link';
import Image from 'next/image';

import maffinLogo from '@/assets/images/maffin_logo_sm.png';
import Menu from './Menu';

export default function LeftSideBar(): JSX.Element {
  return (
    <div className="fixed bg-dark-700 text-center h-screen pt-20 w-20 pb-5 z-10">
      <Link href="/" className="fixed block top-0 h-20 w-20 z-2">
        <span className="flex justify-center items-center ml-6 h-20">
          <Image src={maffinLogo} alt="logo" height="65" />
        </span>
      </Link>

      <Menu />
    </div>
  );
}
