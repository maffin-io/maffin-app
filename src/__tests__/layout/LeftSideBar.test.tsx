import React from 'react';
import { render } from '@testing-library/react';
import type { LinkProps } from 'next/link';

import LeftSidebar from '@/layout/LeftSidebar';

jest.mock('next/link', () => jest.fn(
  (
    props: LinkProps & { children?: React.ReactNode } & React.HTMLAttributes<HTMLAnchorElement>,
  ) => (
    <a className={props.className} href={props.href.toString()}>{props.children}</a>
  ),
));

jest.mock('next/navigation', () => ({
  usePathname: jest.fn(),
}));
// eslint-disable-next-line @typescript-eslint/no-var-requires
const usePathname = jest.spyOn(require('next/navigation'), 'usePathname');

describe('LeftSidebar', () => {
  it('renders as expected', () => {
    usePathname.mockReturnValue('/dashboard/accounts');
    const { container } = render(<LeftSidebar />);

    expect(container).toMatchSnapshot();
  });
});
