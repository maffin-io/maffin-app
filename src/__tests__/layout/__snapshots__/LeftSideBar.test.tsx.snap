// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftSidebar renders as expected 1`] = `
<div>
  <div
    class="fixed bg-dark-700 text-center h-screen pt-20 w-20 pb-5 z-10"
  >
    <a
      class="fixed block top-0 h-20 w-20 z-2"
      href="/"
    >
      <span
        class="flex justify-center items-center h-20"
      >
        <img
          alt="logo"
          data-nimg="1"
          decoding="async"
          height="45"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="45"
        />
      </span>
    </a>
    <ul
      class="flex flex-col h-full"
    >
      <li
        class="group h-12 text-slate-400 hover:bg-cyan-700 hover:text-white hover:w-64 hover:rounded-r-sm"
      >
        <a
          class="flex items-center text-inherit hover:text-inherit h-full px-8 py-4"
          href="/dashboard/accounts"
        >
          <span
            class="mr-8 text-white"
          >
            <svg
              class="text-xl"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 22h14a2 2 0 0 0 2-2v-9a1 1 0 0 0-.29-.71l-8-8a1 1 0 0 0-1.41 0l-8 8A1 1 0 0 0 3 11v9a2 2 0 0 0 2 2zm5-2v-5h4v5zm-5-8.59 7-7 7 7V20h-3v-5a2 2 0 0 0-2-2h-4a2 2 0 0 0-2 2v5H5z"
              />
            </svg>
          </span>
          <span
            class="hidden group-hover:inline-block ml-1"
          >
            Home
          </span>
        </a>
        <span />
      </li>
      <li
        class="group h-12 text-slate-400 hover:bg-cyan-700 hover:text-white hover:w-64 hover:rounded-r-sm"
      >
        <a
          class="flex items-center text-inherit hover:text-inherit h-full px-8 py-4"
          href="/dashboard/investments"
        >
          <span
            class="mr-8 "
          >
            <svg
              class="text-xl"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 3v17a1 1 0 0 0 1 1h17v-2H5V3H3z"
              />
              <path
                d="M15.293 14.707a.999.999 0 0 0 1.414 0l5-5-1.414-1.414L16 12.586l-2.293-2.293a.999.999 0 0 0-1.414 0l-5 5 1.414 1.414L13 12.414l2.293 2.293z"
              />
            </svg>
          </span>
          <span
            class="hidden group-hover:inline-block ml-1"
          >
            Investments
          </span>
        </a>
        <span />
      </li>
      <div
        class="mt-auto"
      >
        <li
          class="group h-12 text-slate-400 hover:bg-cyan-700 hover:text-white hover:w-64 hover:rounded-r-sm"
        >
          <a
            class="flex items-center text-inherit hover:text-inherit h-full px-8 py-4"
            href=""
          >
            <span
              class="mr-8 "
            >
              <svg
                class="text-xl"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 16c2.206 0 4-1.794 4-4s-1.794-4-4-4-4 1.794-4 4 1.794 4 4 4zm0-6c1.084 0 2 .916 2 2s-.916 2-2 2-2-.916-2-2 .916-2 2-2z"
                />
                <path
                  d="m2.845 16.136 1 1.73c.531.917 1.809 1.261 2.73.73l.529-.306A8.1 8.1 0 0 0 9 19.402V20c0 1.103.897 2 2 2h2c1.103 0 2-.897 2-2v-.598a8.132 8.132 0 0 0 1.896-1.111l.529.306c.923.53 2.198.188 2.731-.731l.999-1.729a2.001 2.001 0 0 0-.731-2.732l-.505-.292a7.718 7.718 0 0 0 0-2.224l.505-.292a2.002 2.002 0 0 0 .731-2.732l-.999-1.729c-.531-.92-1.808-1.265-2.731-.732l-.529.306A8.1 8.1 0 0 0 15 4.598V4c0-1.103-.897-2-2-2h-2c-1.103 0-2 .897-2 2v.598a8.132 8.132 0 0 0-1.896 1.111l-.529-.306c-.924-.531-2.2-.187-2.731.732l-.999 1.729a2.001 2.001 0 0 0 .731 2.732l.505.292a7.683 7.683 0 0 0 0 2.223l-.505.292a2.003 2.003 0 0 0-.731 2.733zm3.326-2.758A5.703 5.703 0 0 1 6 12c0-.462.058-.926.17-1.378a.999.999 0 0 0-.47-1.108l-1.123-.65.998-1.729 1.145.662a.997.997 0 0 0 1.188-.142 6.071 6.071 0 0 1 2.384-1.399A1 1 0 0 0 11 5.3V4h2v1.3a1 1 0 0 0 .708.956 6.083 6.083 0 0 1 2.384 1.399.999.999 0 0 0 1.188.142l1.144-.661 1 1.729-1.124.649a1 1 0 0 0-.47 1.108c.112.452.17.916.17 1.378 0 .461-.058.925-.171 1.378a1 1 0 0 0 .471 1.108l1.123.649-.998 1.729-1.145-.661a.996.996 0 0 0-1.188.142 6.071 6.071 0 0 1-2.384 1.399A1 1 0 0 0 13 18.7l.002 1.3H11v-1.3a1 1 0 0 0-.708-.956 6.083 6.083 0 0 1-2.384-1.399.992.992 0 0 0-1.188-.141l-1.144.662-1-1.729 1.124-.651a1 1 0 0 0 .471-1.108z"
                />
              </svg>
            </span>
            <span
              class="hidden group-hover:inline-block ml-1"
            >
              Tools
            </span>
          </a>
          <ul
            class="hidden float-right w-44 bg-dark-700 shadow-md group-hover:inline-block"
          >
            <li
              class="text-sm ml-2 text-left py-2"
            >
              <a
                class="text-slate-400 hover:text-white"
                href="/dashboard/commodities"
              >
                Commodities
              </a>
            </li>
          </ul>
        </li>
        <li
          class="group h-12 text-slate-400 hover:bg-cyan-700 hover:text-white hover:w-64 hover:rounded-r-sm"
        >
          <a
            class="flex items-center text-inherit hover:text-inherit h-full px-8 py-4"
            href="https://blog.maffin.io/docs"
          >
            <span
              class="mr-8 "
            >
              <svg
                class="text-xl"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6 22h15v-2H6.012C5.55 19.988 5 19.805 5 19s.55-.988 1.012-1H21V4c0-1.103-.897-2-2-2H6c-1.206 0-3 .799-3 3v14c0 2.201 1.794 3 3 3zM5 8V5c0-.805.55-.988 1-1h13v12H5V8z"
                />
                <path
                  d="M8 6h9v2H8z"
                />
              </svg>
            </span>
            <span
              class="hidden group-hover:inline-block ml-1"
            >
              Docs
            </span>
          </a>
          <span />
        </li>
      </div>
    </ul>
  </div>
</div>
`;
